src/Univapay/Resources/ThreeDS.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Univapay\Resources;
+
+use Univapay\Enums\ThreeDSMode;
+use Univapay\Utility\FormatterUtils;
+use Univapay\Utility\Json\JsonSchema;
+
+class ThreeDS
+{
+    use Jsonable;
+
+    public $redirectEndpoint;
+    public $redirectId;
+    public $mode;
+
+    public function __construct(
+        $redirectEndpoint,
+        $redirectId,
+        $mode
+    ) {
+        $this->redirectEndpoint = $redirectEndpoint;
+        $this->redirectId = $redirectId;
+        $this->mode = $mode;
+    }
```

Comment:
Add external 3DS MPI fields, please refer to the relevant PR on the API repo. The MPI fields must be all or nothing, so please add the validation into the constructor. Might be a good idea to create a separate obj for the MPI fields and flatten the output in `jsonSerialize`

Reply:
@kgston while i do understand the benefit to implement validation on SDK side, shouldn't SDK just pass it along MPI fields whatever they have been provided with to API and let the API decide if it valid or not, as in the future the validation might change (eg: message_version only accept `2.1.0` and `2.2.0`)

Reply:
As discussed, we just want to ensure that the MPI fields are all or nothing when passing it in. So if the merchant chooses to pass in the MPI fields, we just need a validation in the constructor that all fields are populated. Otherwise, the MPI object should be null in the `PaymentThreeDS` class.